File: tests/test_config_swarm.py
# test_config_swarm.py
"""
Tests for Swarm configuration - critical for multi-LLM consensus functionality.
Verifies that swarm settings are properly stored and persisted.
"""

import unittest
import json
import os
from pathlib import Path
from config_system import AppConfig


class TestConfigSwarm(unittest.TestCase):
    """TestConfigSwarm class."""

    def setUp(self):
        self.test_config_path = Path("test_config.json")
        if self.test_config_path.exists():
            os.remove(self.test_config_path)

    def tearDown(self):
        if not self.test_config_path.exists():
            return

        os.remove(self.test_config_path)

    def test_swarm_defaults(self):
        """Verify that swarm settings have correct defaults."""
        config = AppConfig()
        # These are critical for multi-LLM consensus functionality
        self.assertTrue(hasattr(config, "SWARM_SIZE"), "AppConfig missing SWARM_SIZE")
        self.assertTrue(hasattr(config, "SWARM_ENABLED"), "AppConfig missing SWARM_ENABLED")
        self.assertEqual(config.SWARM_SIZE, 3)  # Default swarm size
        self.assertEqual(config.SWARM_ENABLED, False)  # Disabled by default

    def test_swarm_attributes_sync(self):
        """Verify uppercase and lowercase swarm attributes are in sync."""
        config = AppConfig()
        # Both uppercase and lowercase should work
        self.assertEqual(config.swarm_enabled, config.SWARM_ENABLED)
        self.assertEqual(config.swarm_size, config.SWARM_SIZE)

    def test_swarm_config_modification(self):
        """Verify swarm config can be modified."""
        config = AppConfig()
        # Modify lowercase version
        config.swarm_enabled = True
        config.swarm_size = 5
        # Check values
        self.assertTrue(config.swarm_enabled)
        self.assertEqual(config.swarm_size, 5)


if __name__ == "__main__":
    unittest.main()

File: rust_output/src/test_config_swarm.rs
/// Tests for Swarm configuration - critical for multi-LLM consensus functionality.
/// Verifies that swarm settings are properly stored and persisted.

use anyhow::{Result, Context};
use crate::config_system::{AppConfig};
use std::path::PathBuf;

/// TestConfigSwarm class.
#[derive(Debug, Clone)]
pub struct TestConfigSwarm {
}

impl TestConfigSwarm {
    pub fn setUp(&mut self) -> () {
        self.test_config_path = PathBuf::from("test_config.json".to_string());
        if self.test_config_path.exists() {
            std::fs::remove_file(self.test_config_path).ok();
        }
    }
    pub fn tearDown(&self) -> () {
        if !self.test_config_path.exists() {
            return;
        }
        std::fs::remove_file(self.test_config_path).ok();
    }
    /// Verify that swarm settings have correct defaults.
    pub fn test_swarm_defaults(&mut self) -> () {
        // Verify that swarm settings have correct defaults.
        let mut config = AppConfig();
        assert!(/* hasattr(config, "SWARM_SIZE".to_string()) */ true, "AppConfig missing SWARM_SIZE".to_string());
        assert!(/* hasattr(config, "SWARM_ENABLED".to_string()) */ true, "AppConfig missing SWARM_ENABLED".to_string());
        assert_eq!(config::SWARM_SIZE, 3);
        assert_eq!(config::SWARM_ENABLED, false);
    }
    /// Verify uppercase and lowercase swarm attributes are in sync.
    pub fn test_swarm_attributes_sync(&mut self) -> () {
        // Verify uppercase and lowercase swarm attributes are in sync.
        let mut config = AppConfig();
        assert_eq!(config::swarm_enabled, config::SWARM_ENABLED);
        assert_eq!(config::swarm_size, config::SWARM_SIZE);
    }
    /// Verify swarm config can be modified.
    pub fn test_swarm_config_modification(&mut self) -> () {
        // Verify swarm config can be modified.
        let mut config = AppConfig();
        config::swarm_enabled = true;
        config::swarm_size = 5;
        assert!(config::swarm_enabled);
        assert_eq!(config::swarm_size, 5);
    }
}
